### PR TITLE
Workout

### PR DESCRIPTION
**Pull request by regilkent123**

### CODE REVIEW

**Comment by ozknightwalker** [comment]
@regilkent123 @iamkatrinaanne please fix merge conflict

File: OJT/urls.py [hunk 1]
```diff
@@ -14,8 +14,16 @@
     2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
 """
 from django.contrib import admin
+from django.conf.urls import url
 from django.urls import path
+from django.urls import path, include
```
Comment (ozknightwalker) [inline OJT/urls.py]: @regilkent123 @iamkatrinaanne this is imported above please make this import in one line

File: manage.py [hunk 2]
```diff
@@ -12,4 +13,4 @@
             "available on your PYTHONPATH environment variable? Did you "
             "forget to activate a virtual environment?"
         ) from exc
-    execute_from_command_line(sys.argv)
+    execute_from_command_line(sys.argv)
```
Comment (ozknightwalker) [inline manage.py:15]: @regilkent123 @iamkatrinaanne please remove this changes, you guys don't need to modify this file

File: userprofile/models.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+from django.contrib.auth.models import User
+
+class UserProfile(models.Model):
+    user = models.OneToOneField(
+            User, on_delete=models.CASCADE,
+            primary_key=True,
+        )
+
+    trainee = 1
+    trainer = 2
```
Comment (ozknightwalker) [inline userprofile/models.py]: @regilkent123 @iamkatrinaanne please make static variables all caps

File: userprofile/views.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
+from django.shortcuts import render
+from django.views import generic
```
Comment (ozknightwalker) [inline userprofile/views.py]: @regilkent123 @iamkatrinaanne you don't need to import this anymore

File: userprofile/views.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+
+    def get(self, request, *args, **kwargs):
+        form = self.form_class(initial=self.initial)
+        return render(request, self.template_name, {'form': form})
+
+
+    def post(self,request, *args, **kwargs):
+        form = self.form_class(request.POST)
+        print (form.is_valid(), form.errors)
```
Comment (ozknightwalker) [inline userprofile/views.py]: @regilkent123 @iamkatrinaanne please don't commit any code with print statements

File: userprofile/views.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        if form.is_valid():
+            form.save()
+            username = form.cleaned_data['username']
+            raw_password = form.cleaned_data['password']
+            user = authenticate(username=username, password=raw_password)
+            if user is not None:
```
Comment (ozknightwalker) [inline userprofile/views.py]: @regilkent123 @iamkatrinaanne what happens when user failed to authenticate?

File: userprofile/views.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+                return HttpResponseRedirect('/afterlogin/')
+        return render(request, self.template_name, {'form': form})
+
+
+
+class LoginView(generic.TemplateView):
+    template_name = 'userprofile/login.html'
+
+
+def auth_login(request):
```
Comment (ozknightwalker) [inline userprofile/views.py]: @regilkent123 @iamkatrinaanne please use classedbased views

File: workout/forms.py [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+        fields = ['workout_name', 'workout_type', 'description', 'workout_photo']
+        widgets = {
+            'description': forms.Textarea(attrs={'cols': 10, 'rows': 20}),
+        }
+
+    def __init__(self, *args, **kwargs):
+        super(WorkoutForm, self).__init__(*args, **kwargs)
+        self.fields['workout_type'].empty_label =  None
+
+    def save(self):
```
Comment (ozknightwalker) [inline workout/forms.py]: @regilkent123 @regilkent123 this is an incorrect way of overriding form save. http://astronautradio.com/post/16186725117/how-to-overwrite-the-save-of-a-django-form-like-a or research

File: workout/migrations/0001_initial.py [hunk 9]
```diff
@@ -0,0 +1,23 @@
+# Generated by Django 2.0.4 on 2018-05-16 06:04
+
```
Comment (ozknightwalker) [inline workout/migrations/0001_initial.py]: @regilkent123 @iamkatrinaanne merge your migrations to 1 file

File: workout/models.py [hunk 10]
```diff
@@ -0,0 +1,22 @@
+from django.db import models
+
+class Workout(models.Model):
+    workout_name = models.CharField(max_length = 200)
+    aerobic = 1
```
Comment (ozknightwalker) [inline workout/models.py]: @regilkent123 @iamkatrinaanne make static variables all in  CAPS

File: workout/models.py [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+    strength = 2
+    balance = 3
+    flexibility = 4
+    workoutType = (
+        (aerobic, 'Aerobic exercise'),
+        (strength, 'Strength exercise'),
+        (balance, 'Balance exercise'),
+        (flexibility, 'Flexibility exercise'),
+        )
+    workout_type = models.IntegerField(choices = workoutType, blank = False, default = None)
```
Comment (ozknightwalker) [inline workout/models.py]: @regilkent123 @iamkatrinaanne use positive integerfield here

Comment (ozknightwalker) [reply workout/models.py]: also don't put spaces around `=` if its an argument and not a statement. please review reading coding with python :)

**Review by ozknightwalker — CHANGES_REQUESTED** [review]

File: manage.py (hunk 2 above)
Comment (regilkent123) [reply manage.py:15]: i dont remember changing something on this kuya @ozknightwalker

Comment (ozknightwalker) [reply manage.py:15]: @regilkent123 @iamkatrinaanne https://github.com/regilkent123/OJTPROJECT/blame/d41dae1df9e71dfb12343dc380a9846262783d93/manage.py :)

Comment (ozknightwalker) [reply manage.py:15]: added spaces around the code block

File: workout/forms.py (hunk 8 above)
Comment (regilkent123) [reply workout/forms.py]: why sayop kuya? @ozknightwalker

**Review by ozknightwalker — APPROVED** [review]

**Comment by ozknightwalker** [comment]
lgtm

**Pull request merged**